File: app/src/main/java/com/example/recyclerview/MyRecyclerViewAdapter.java

package com.example.recyclerview;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.graphics.Color;
import android.os.Build;
import android.view.ActionMode;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;
import com.example.recyclerview.Model.Tasks;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import java.util.HashMap;

import io.realm.OrderedRealmCollection;
import io.realm.RealmRecyclerViewAdapter;


public class MyRecyclerViewAdapter extends RealmRecyclerViewAdapter<Tasks,MyRecyclerViewAdapter.MyViewHolder> {

    MainActivity mainActivity;
    OrderedRealmCollection<Tasks> data;
    private FirebaseDatabase db=FirebaseDatabase.getInstance();
    private DatabaseReference root=db.getReference();
    public MyRecyclerViewAdapter(OrderedRealmCollection<Tasks> data) {
        super(data,true);
        this.data=data;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull  ViewGroup parent, int viewType) {


        View view= LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_layout,parent,false);
        return new MyViewHolder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull  MyRecyclerViewAdapter.MyViewHolder holder, int position) {


        final Tasks tasks = getItem(position);

        holder.idTV.setText(String.valueOf(tasks.getTask_id()));
        holder.tasknameTV.setText(tasks.getTask_name());

        holder.icondelete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                AlertDialog.Builder builder=new AlertDialog.Builder(v.getRootView().getContext());
                View dialogview=LayoutInflater.from(v.getRootView().getContext()).inflate(R.layout.custom_dialog,null);

                builder.setView(dialogview);
                builder.setCancelable(true);

                builder.setNegativeButton("Cancel", null);
                builder.setPositiveButton("Delete", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                        RealMHelper helper = new RealMHelper();
                        helper.deleteData(data.get(position).getTask_id());

                    }
                });
                builder.show();

            }
        });



        holder.like.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {

                HashMap<String, String>userMap=new HashMap<>();
                userMap.put("Tasks", tasks.getTask_name());
                root.push().setValue(userMap);
                Toast.makeText(v.getContext(), "Data saved succesfully...",Toast.LENGTH_LONG).show();
            }
        });


    }


    @Override
    public long getItemId(int position) {
        return getItem(position).getTask_id();

    }

    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView idTV,tasknameTV;
        Button like,icondelete;
        public MyViewHolder( View itemView) {
            super(itemView);
            idTV=itemView.findViewById(R.id.idTV);
            tasknameTV=itemView.findViewById(R.id.tasknameTV);
            like=itemView.findViewById(R.id.like);
            icondelete=itemView.findViewById(R.id.icondelete);



        }

    }
}

File: app/src/main/java/com/example/recyclerview/Adapter/MyRecyclerViewAdapter.kt
package com.example.recyclerview.Adapter

import android.app.AlertDialog
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.example.recyclerview.Model.Tasks
import com.example.recyclerview.Adapter.MyRecyclerViewAdapter.MyViewHolder
import com.example.recyclerview.UI.MainActivity
import com.example.recyclerview.R
import com.example.recyclerview.UI.RealMHelper
import com.google.firebase.database.FirebaseDatabase
import io.realm.OrderedRealmCollection
import io.realm.RealmRecyclerViewAdapter
import java.util.*

class MyRecyclerViewAdapter(data: OrderedRealmCollection<Tasks?>) :
    RealmRecyclerViewAdapter<Tasks?, MyViewHolder>(
        data, true
    ) {
    var mainActivity: MainActivity? = null
    private val db = FirebaseDatabase.getInstance()
    private val root = db.reference
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.rv_layout, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val tasks = getItem(position)
        holder.idTV.text = java.lang.String.valueOf(tasks!!.task_id)
        holder.tasknameTV.text = tasks.task_name
        holder.icondelete.setOnClickListener { v ->
            val builder = AlertDialog.Builder(v.rootView.context)

            //ayrı bi classa ekle

            val dialogview =
                LayoutInflater.from(v.rootView.context).inflate(R.layout.custom_dialog, null)
            builder.setView(dialogview)
            builder.setCancelable(true)
            builder.setNegativeButton("Cancel", null)
            builder.setPositiveButton("Delete") { dialogInterface, i ->
                val helper = RealMHelper()
                data?.get(position)?.let { helper.deleteData(it.task_id) }
            }
            builder.show()
        }
        holder.like.setOnClickListener { v ->
            val userMap = HashMap<String, String?>()
            userMap["Tasks"] = tasks.task_name
            root.push().setValue(userMap)
            Toast.makeText(v.context, "Data saved succesfully...", Toast.LENGTH_LONG).show()
        }
    }

    override fun getItemId(position: Int): Long {
        return getItem(position)!!.task_id.toLong()
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var idTV: TextView
        var tasknameTV: TextView
        var like: Button
        var icondelete: Button

        init {
            idTV = itemView.findViewById(R.id.idTV)
            tasknameTV = itemView.findViewById(R.id.tasknameTV)
            like = itemView.findViewById(R.id.like)
            icondelete = itemView.findViewById(R.id.icondelete)
        }
    }
}